src/core/shouldUpdateReactComponent.js

```diff
@@ -63,19 +63,30 @@ function shouldUpdateReactComponent(prevElement, nextElement) {
             if (typeof nextElement.type !== 'string' ||
                 nextElement.type === 'input' ||
                 nextElement.type === 'textarea') {
-              warning(
-                false,
-                '<%s /> is being rendered by both %s and %s using the same key ' +
-                '(%s) in the same place. Currently, this means that they ' +
-                'don\'t preserve state. This behavior should be very rare ' +
-                'so we\'re considering deprecating it. Please contact the ' +
-                'React team and explain your use case so that we can take that ' +
-                'into consideration.',
-                nextDisplayName || 'Unknown Component',
-                prevName || '[Unknown]',
-                nextName || '[Unknown]',
-                prevElement.key
-              );
+              if ((prevElement._owner != null &&
+                  prevElement._owner._isOwnerNecessary === false) ||
+                  (nextElement._owner != null &&
+                  nextElement._owner._isOwnerNecessary === false)) {
+                if (prevElement._owner != null) {
+                  prevElement._owner._isOwnerNecessary = true;
+                }
+                if (nextElement._owner != null) {
+                  nextElement._owner._isOwnerNecessary = true;
+                }
+                warning(
+                  false,
+                  '<%s /> is being rendered by both %s and %s using the same key ' +
+                  '(%s) in the same place. Currently, this means that they ' +
+                  'don\'t preserve state. This behavior should be very rare ' +
+                  'so we\'re considering deprecating it. Please contact the ' +
+                  'React team and explain your use case so that we can take that ' +
+                  'into consideration.',
```

Comment:
Arguably this is a perfect example of why you should avoid statement nesting. :)

> On Feb 13, 2015, at 6:00 PM, Jim <notifications@github.com> wrote:
> 
> In src/core/shouldUpdateReactComponent.js:
> 
> > +                  (nextElement._owner != null &&
> > +                  nextElement._owner._isOwnerNecessary === false)) {
> > +                if (prevElement._owner != null) {
> > +                  prevElement._owner._isOwnerNecessary = true;
> > +                }
> > +                if (nextElement._owner != null) {
> > +                  nextElement._owner._isOwnerNecessary = true;
> > +                }
> > +                warning(
> > +                  false,
> > +                  '<%s /> is being rendered by both %s and %s using the same key ' +
> > +                  '(%s) in the same place. Currently, this means that they ' +
> > +                  'don\'t preserve state. This behavior should be very rare ' +
> > +                  'so we\'re considering deprecating it. Please contact the ' +
> > +                  'React team and explain your use case so that we can take that ' +
> > +                  'into consideration.',
> Rrrarw, yeah, ok. This is a perfect example of why the line length warning SUCKS and should be removed.
> 
> —
> Reply to this email directly or view it on GitHub.
>